pkg/tiller/release_server.go

```diff
@@ -252,6 +252,15 @@ func (s *ReleaseServer) renderResources(ch *chart.Chart, values chartutil.Values
 		return nil, nil, "", fmt.Errorf("Chart incompatible with Tiller %s", sver)
 	}
 
+	if ch.Metadata.KubeVersion != "" {
+		cap, _ := values["Capabilities"].(*chartutil.Capabilities)
+		gitVersion := cap.KubeVersion.String()
+		k8sVersion := strings.Split(gitVersion, "+")[0]
+		if !version.IsCompatibleRange(ch.Metadata.KubeVersion, k8sVersion) {
+			return nil, nil, "", fmt.Errorf("Chart incompatible with Kubernetes %s", k8sVersion)
```

Comment:
`Chart incompatible with Kubernetes %s", k8sVersion` might be better as something like `Chart requires kubernetesVersion: %s which is incompatible with Kubernetes %s", ch.Metadata.KubeVersion, k8sVersion`
I feel this would be more helpful for new users as the current error doesn't indicate why its not compatible. The test for the result string would need to be updated.